### ISSUE TITLE
Permission Access with PackageCompiler.force_native_image!()

### ISSUE THREAD

**Issue by CoreyShupe** [JuliaLang/PackageCompiler.jl#108]
My OS is arch-linux
I'm running `PackageCompiler.force_native_image!()` as instructed on first run, however it's spitting back a permission error.
Is there currently a way to fix default write permissions of generated files?
error:
<br />
```
julia> PackageCompiler.force_native_image!()
ERROR: open: permission denied (EACCES)
Stacktrace:
 [1] uv_error at ./libuv.jl:68 [inlined]
 [2] open(::String, ::UInt16, ::UInt16) at ./filesystem.jl:81
 [3] sendfile(::String, ::String) at ./file.jl:523
 [4] #cp#10(::Bool, ::Bool, ::Function, ::String, ::String) at ./file.jl:227
 [5] (::Base.Filesystem.#kw##cp)(::Array{Any,1}, ::Base.Filesystem.#cp, ::String, ::String) at ./<missing>:0
 [6] rename(::String, ::String) at ./file.jl:510
 [7] (::Base.Filesystem.#kw##mv)(::Array{Any,1}, ::Base.Filesystem.#mv, ::String, ::String) at ./<missing>:0
 [8] copy_system_image(::String, ::String, ::Bool) at /home/effectz/.julia/v0.6/PackageCompiler/src/PackageCompiler.jl:48
 [9] force_native_image!() at /home/effectz/.julia/v0.6/PackageCompiler/src/api.jl:116
 [10] macro expansion at ./REPL.jl:97 [inlined]
 [11] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73
```
info:
<br />
```
julia> versioninfo()
Julia Version 0.6.4
Commit 9d11f62bcb (2018-07-09 19:09 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i3-4340 CPU @ 3.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Haswell MAX_THREADS=16)
  LAPACK: libopenblas
  LIBM: libm
  LLVM: libLLVM-3.9.1 (ORCJIT, haswell)
```

**Comment by x-ji** [comment]
Though I was able to successfully run `PackageCompiler.force_native_image!()` if I simply run `sudo julia`... It then was able to:

```
/usr/share/julia/base/precompile.jl
Build shared library "/root/.julia/v0.6/PackageCompiler/sysimg/backup/native/sys.so":
  `cc -shared -o /root/.julia/v0.6/PackageCompiler/sysimg/backup/native/sys.so /root/.julia/v0.6/PackageCompiler/sysimg/backup/native/sys.o -std=gnu99 -I/usr/include/julia -DJULIA_ENABLE_THREADING=1 -fPIC -L/usr/lib -Wl,--export-dynamic -Wl,-rpath,/usr/lib -Wl,-rpath,/usr/lib/julia -ljulia -m64`
INFO: Copying system image: /root/.julia/v0.6/PackageCompiler/sysimg/backup/native/sys.so to /usr/lib/julia/sys.so
```

Not sure if that's the right approach though.

**Comment by jg110** [comment]
I've got this error as well on Julia 1.0.1. After this failure I was unable to use the Dierckx package that I had previously installed. I assume I would also have been prevented from using any other package. Julia suggested using Pkg to recompile Dierckx, but I got a permission denied error when I tried. Strangely, I (as a non-root user) was still the owner of all files within the ~/.julia directory, and permissions seemed to be set correctly to allow programs I launched to modify files within the directory.
I uninstalled Julia, deleted the .julia directory, and reinstalled Julia to get it usable again.
